src/odemis/driver/xt_client.py

```diff
@@ -2215,3 +2237,90 @@ def _setMultiBeamMode(self, multi_beam_mode):
+        except Exception as err:
+            raise HwError("Failed to connect to XT server '%s'. Check that the "
+                          "uri is correct and XT server is"
+                          " connected to the network. %s" % (address, err))
+        Detector.__init__(self, name, role, parent=parent, channel_name=channel_name, **kwargs)
+        self._shape = (2**16,)  # Depth of the image
+
+    def stop_generate(self):
+        logging.debug("Stopping image acquisition")
+        with self._proxy_access:
+            self.cancel_connection._pyroClaimOwnership()
+            # Directly calling set_channel_state does not work with XT. Therefore
+            # call get_channel_state, before trying to stop the channel.
+            self.cancel_connection.get_channel_state(self._channel_name)
+            self.cancel_connection.set_channel_state(self._channel_name, False)
+            # Stop twice, to make sure the channel fully stops.
+            self.cancel_connection.set_channel_state(self._channel_name, False)
+        if self.parent._scanner.blanker.value is None:
+            self.parent.blank_beam()
+        self._genmsg.put(GEN_STOP)
+
+    def _acquire(self):
+        """
+        Acquisition thread
+        Managed via the ._genmsg Queue
+        """
+        try:
+            while True:
+                # Wait until we have a start (or terminate) message
+                self._acq_wait_start()
+                logging.debug("Preparing acquisition")
+                while True:
+                    if self._acq_should_stop():
+                        break
+
+                    # Start a complete scan.
+                    try:
+                        self.parent.scan_image()
+                    except OSError as err:
+                        if err.errno == -2147467260:
+                            logging.debug("Scan image aborted.")
+                            # Operation aborted, indicating acquisition should stop.
+                            break
+                        else:
+                            raise
+
+                    # Acquire the image
+                    # =================
+                    image = self.parent.get_latest_image(self._channel_name)
+
+                    md = self.parent._scanner._metadata.copy()
+                    md.update(self._metadata)
+                    md[model.MD_DWELL_TIME] = self.parent._scanner.dwellTime.value
+                    md[model.MD_ROTATION] = self.parent._scanner.rotation.value
+
+                    da = DataArray(image, md)
+                    logging.debug("Notify dataflow with new image.")
+                    self.data.notify(da)
+            logging.debug("Acquisition stopped")
+        except TerminationRequested:
+            logging.debug("Acquisition thread requested to terminate")
+        except Exception as err:
+            logging.exception("Failure in acquisition thread: {}".format(err))
```

Comment:
In logging, please just pass the values as argument `"Failure in acquisition thread: %s", err`. Mostly to keep the code homogeneous.